code/modules/power/supermatter/supermatter.dm

```diff
@@ -819,6 +833,106 @@
+		factor = clamp(damage/600, 1, 10),
+		density = clamp(damage/10, 12, 100)
+	))
+
+	// Filter animation persists even if the filter itself is changed externally.
+	// Probably prone to breaking. Treat with suspicion.
+	if(new_filter)
+		animate(get_filter("ray"), offset = 10, time = 10 SECONDS, loop = -1)
+		animate(offset = 0, time = 10 SECONDS)
+
+	if(power > POWER_PENALTY_THRESHOLD)
+		add_filter(name = "ray", priority = 1, params = list(
+			type = "rays",
+			size = power ? clamp((damage/100) * power, 50, 175) : 1, //Higher power
+			color = SUPERMATTER_TESLA_COLOUR,
+			factor = clamp(damage/300, 1, 20),
+			density = clamp(damage/5, 12, 200)
+		))
+		if(prob(25))
+			new /obj/effect/warp_effect/bsg(get_turf(src)) //Some extra visual effect to the shocking sm which is a bit less interesting.
+		if(final_countdown)
+			add_filter(name = "icon", priority = 2, params = list(
+				type = "layer",
+				icon = new/icon('icons/obj/tesla_engine/energy_ball.dmi', "energy_ball", frame = rand(1,12)),
+				flags = FILTER_UNDERLAY
+			))
+		else
+			remove_filter("icon")
+
+	if(combined_gas > MOLE_PENALTY_THRESHOLD)
+		add_filter(name = "ray", priority = 1, params=list(
+			type = "rays",
+			size = power ? clamp((damage/100) * power, 50, 125) : 1,
+			color = SUPERMATTER_SINGULARITY_RAYS_COLOUR,
+			factor = clamp(damage / 300, 1, 30),
+			density = clamp(damage / 5, 12, 300) //More compressed due to gravity
+		))
+
+		add_filter(name = "outline", priority = 2, params = list(
+			type = "outline",
+			size = 1,
+			color = SUPERMATTER_SINGULARITY_LIGHT_COLOUR
+		))
+		if(!warp)
+			warp = new(src)
+			vis_contents += warp
+		animate(warp, time = 6, transform = matrix().Scale(0.5,0.5))
+		animate(time = 14, transform = matrix())
+		if(final_countdown)
+			add_filter(name = "icon", priority = 3, params = list(
+				type = "layer",
+				icon = new/icon('icons/effects/96x96.dmi', "singularity_s3", frame = rand(1,8)),
+				flags = FILTER_OVERLAY
+			))
+		else
+			remove_filter("icon")
+	else
+		vis_contents -= warp
+		remove_filter("outline")
+		QDEL_NULL(warp)
+
+
+
+// Change how bright the rock is
+/obj/machinery/atmospherics/supermatter_crystal/proc/lights()
+	set_light(
+		l_range = 4 + power / 200,
+		l_power = 1 + power / 1000,
+		l_color = gasmix_power_ratio > 0.8 ? SUPERMATTER_RED : SUPERMATTER_COLOUR,
+	)
+
+	if(power > POWER_PENALTY_THRESHOLD)
+		set_light(
+			l_range = 4 + clamp(damage * power, 50, 500),
+			l_power = 3,
+			l_color = SUPERMATTER_TESLA_COLOUR,
+		)
+	if(combined_gas > MOLE_PENALTY_THRESHOLD)
+		set_light(
+			l_range = 4 + clamp(damage / 2, 10, 50),
+			l_power = 3,
+			l_color = SUPERMATTER_SINGULARITY_LIGHT_COLOUR,
+		)
+
+
+/obj/effect/warp_effect/supermatter
+	plane = GRAVITY_PULSE_PLANE
+	appearance_flags = PIXEL_SCALE|LONG_GLIDE // no tile bound so you can see it around corners and so
+	icon = 'icons/effects/light_352.dmi'
+	icon_state = "light"
+	pixel_x = -176
+	pixel_y = -176
```

Comment:
Why on earth are these pixel vars set to this? 👀

Reply:
The icon is 352 pixels wide(thus light_352). As such, this centers the image.

Reply:
Makes sense.